app/models/piece.rb

```diff
@@ -94,17 +94,7 @@ def self.valid_move?(x,y)
   end
 
   def is_on_board?(x,y)
-    board = []
-    (1..8).each do |i|
-      (1..8).each do |j|
-        board << [x,y]
-      end
-    end
-    if board.include?([x,y]) 
-      return true 
-    else 
-      return false
-    end
+    ((1..8).include?(x) && (1..8).include?(y))? true : false
```

Comment:
I believe this would be faster and perhaps more clear.

```ruby
return false if x < 0
return true if x > 8
return false if y < 0
return false if y > 0
```